src/adam7.rs

```diff
@@ -6,6 +6,42 @@
 /// applies to a decoded row.
 ///
 /// See also [Reader.next_interlaced_row](crate::decoder::Reader::next_interlaced_row).
+use std::cmp::min;
+
+struct PassConstants {
+    line_mul: usize,
+    line_off: usize,
+    samp_mul: usize,
+    samp_off: usize,
```

Comment:
Rust convention is to mostly avoid abbreviations in variable names. So for instance, `sample_offset` rather than `samp_off`